### PR TITLE
Refactor: make Vizier kwargs keyword-only

### PR DESCRIPTION
**Pull request by keflavich**
See #1746.

This refactor raises a question: should we keyword-only the "private" functions too?

### CODE REVIEW

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Hahh, was about to push the same changes 😄 

I go ahead and merge this, so the new PR can be on top of this.

**Comment by keflavich** [comment]
There's a bug in the docs that I was working on - should I put this down for now?

File: astroquery/vizier/core.py [hunk 1]
```diff
@@ -332,6 +332,8 @@ def query_region_async(self, coordinates, radius=None, inner_radius=None,
         Serves the same purpose as `query_region` but only
         returns the HTTP response rather than the parsed result.
 
+        At least one of `radius` or `width` must be specified.
```
Comment (bsipocz) [inline astroquery/vizier/core.py:335]:
```suggestion
        At least one of ``radius`` or ``width`` must be specified.
```

**Review by bsipocz — APPROVED** [review]
Minor fix for sphinx

**Comment by bsipocz** [comment]
> There's a bug in the docs that I was working on - should I put this down for now?

I didn't touch the docs, we don't yet have it included in the tests, so it doesn't come up as a failure.

**Comment by keflavich** [comment]
I meant the docstr fix you suggested - I pushed it before I saw your suggestion =)

**Comment by keflavich** [comment]
What about changelog?  Should we individually changelog these?

**Comment by bsipocz** [comment]
Yes, changelog is needed, but I just go ahead with the merge and include its changelog in the new PR.

**Pull request merged**